sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -736,7 +741,6 @@ qualifiedName
 
 identifier
     : strictIdentifier
-    | {ansi}? ansiReserved
```

Comment:
This is the only code change I expect. What's the rationale of the other changes? I think `select current_date()` should not be supported in ansi mode.

Reply:
or, we can find some more examples to advocate it. AFAIK presto supports `select current_date()` although it's not SQL standard. Do you know of any other systems that support it?

Reply:
yeah, it seems `current_timestamp()` is not a topic of the ANSI standard. So, I'll drop it in this pr.

But, some databases support `current_timestamp()` and this is implementation-specific.
For example, postgresql/oracle support `current_timestamp(precision)` as follows;
```
postgres=# select CURRENT_TIMESTAMP;
              now              
-------------------------------
 2019-03-12 20:22:52.065108+09
(1 row)

postgres=# select CURRENT_TIMESTAMP(1);
       timestamptz        
--------------------------
 2019-03-12 20:22:56.2+09
(1 row)

postgres=# select CURRENT_TIMESTAMP();
ERROR:  syntax error at or near ")" at character 26
STATEMENT:  select CURRENT_TIMESTAMP();
ERROR:  syntax error at or near ")"
LINE 1: select CURRENT_TIMESTAMP();
                                 ^
```
So, it might be worth supporting this function for better portability even when ansi mode enabled (this is future work though...).

Reply:
As for the ANSI starndard, we need to support these functions below for datetime, too;
```
postgres=# select CURRENT_TIME;
       timetz       
--------------------
 20:33:08.179954+09
(1 row)

postgres=# select LOCALTIME;
      time       
-----------------
 20:33:54.281054
(1 row)

postgres=# select LOCALTIMESTAMP;
         timestamp         
---------------------------
 2019-03-12 20:33:57.85737
(1 row)
```
I'll file a jira later.